app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -198,28 +237,104 @@ class TableVisualization extends Visualization {
-        }
         this.agGridOptions.api.setRowData(rowData)
         this.updateTableSize(this.dom.getAttributeNS(null, 'width'))
     }
 
+    makeOption(value, label) {
+        const optionElem = document.createElement('option')
+        optionElem.value = value
+        optionElem.appendChild(document.createTextNode(label))
+        return optionElem
+    }
+
+    makeButton(label, onclick) {
+        const buttonElem = document.createElement('button')
+        buttonElem.name = label
+        buttonElem.appendChild(document.createTextNode(label))
+        buttonElem.addEventListener('click', onclick)
+        return buttonElem
+    }
+
+    createRowDropdown(all_rows_count, dataTruncated) {
+        const pageLimit = Math.ceil(all_rows_count / this.row_limit)
+        if (this.page > pageLimit) {
+            this.page = pageLimit
+        }
+
+        if (this.statusElem.childElementCount === 0) {
+            this.statusElem.appendChild(
+                this.makeButton('«', () => this.setRowLimitAndPage(this.row_limit, 0))
+            )
+            this.statusElem.appendChild(
+                this.makeButton('‹', () => this.setRowLimitAndPage(this.row_limit, this.page - 1))
+            )
+
+            const selectElem = document.createElement('select')
+            selectElem.name = 'row-limit'
+            selectElem.addEventListener('change', e => {
+                this.setRowLimitAndPage(e.target.value, this.page)
+            })
+            this.statusElem.appendChild(selectElem)
+
+            const rowCountSpanElem = document.createElement('span')
+            this.statusElem.appendChild(rowCountSpanElem)
+
+            this.statusElem.appendChild(
+                this.makeButton('›', () => this.setRowLimitAndPage(this.row_limit, this.page + 1))
+            )
+            this.statusElem.appendChild(
+                this.makeButton('»', () => this.setRowLimitAndPage(this.row_limit, pageLimit - 1))
+            )
+        }
+
+        // Enable/Disable Page buttons
+        this.statusElem.children.namedItem('«').disabled = this.page === 0
+        this.statusElem.children.namedItem('‹').disabled = this.page === 0
+        this.statusElem.children.namedItem('›').disabled = this.page === pageLimit - 1
+        this.statusElem.children.namedItem('»').disabled = this.page === pageLimit - 1
+
+        // Update row limit dropdown and row count
+        const rowCountElem = this.statusElem.getElementsByTagName('span')[0]
+        const rowLimitElem = this.statusElem.children.namedItem('row-limit')
+        if (all_rows_count > 1000) {
+            rowLimitElem.style.display = 'inline-block'
+            const rowCounts = [
+                1000,
+                2500,
+                5000,
+                10000,
+                25000,
+                50000,
+                100000,
+                all_rows_count,
+            ].filter(r => r <= all_rows_count && r <= 100000)
```

Comment:
I'm picky, but if I read this code correctly, if `all_rows_count == 2500`, won't the `2500` entry be duplicated? That would look bad, let's ensure they are unique.